### PR TITLE
arm/irq: move irq_get_guest_info call under spinlock

### PR DESCRIPTION
**Pull request by aanisov**
That function has an intended ASSERT(), but asserts are only
effective for debug builds.

Signed-off-by: Andrii Anisov <andrii_anisov@epam.com>

### CODE REVIEW

File: xen/arch/arm/irq.c [hunk 1]
```diff
@@ -191,12 +191,14 @@ int request_irq(unsigned int irq, unsigned int irqflags,
...
     struct irq_desc *desc = irq_to_desc(gsx_irq_num);
-    struct irq_guest *info = irq_get_guest_info(desc);
+    struct irq_guest *info;
     unsigned long flags;
     int i;
 
     spin_lock_irqsave(&desc->lock, flags);
 
+    info = irq_get_guest_info(desc);
+
```
Comment (lorc) [inline xen/arch/arm/irq.c]: I have no idea why this change help with mentioned assert. Could you please elaborate commit description?

**Comment by aanisov** [comment]
The commit message is reworked, please review.

**Comment by otyshchenko1** [comment]
lgtm
